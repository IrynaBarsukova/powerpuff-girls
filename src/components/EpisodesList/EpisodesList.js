import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import styles from  './EpisodesList.scss';

function EpisodesList({ episodes }) {
    return (
        <div className={styles.episodes}>
            <h3 className={styles.episodes__title}>Episodes</h3>
            <div className={styles.episodesList}>
                {
                    episodes.map(({ id, name, season, number, image }) => (
                        <div key={id} className={styles.episodesList__episode}>
                            <Link to={location => `${location.pathname}/episodes/${id}`}>
                                <img src={image && image.medium} alt={name}/>
                                <p>{name}</p>
                                <p>{season}x{number}</p>
                            </Link>
                        </div>
                    ))
                }
            </div>
        </div>
    );
}

EpisodesList.propTypes = {
    /**
     * The list of show's available episodes
     */
    episodes: PropTypes.array,
};

export default EpisodesList;
