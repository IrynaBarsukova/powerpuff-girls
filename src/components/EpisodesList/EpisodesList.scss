.episodes {
  margin-top: 30px;
}

.episodes__title {
  font-weight: normal;
  margin-bottom: 15px;
}

.episodesList {
  display: flex;
  justify-content: space-between;
  overflow-y: scroll;
}

.episodesList__episode {
  min-width: 250px;

  &:not(:first-child) {
    margin-left: 20px;
  }
}
