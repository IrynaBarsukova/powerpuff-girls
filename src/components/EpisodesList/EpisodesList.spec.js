import React from 'react';

import renderWithRouter from '../../mocks/router.mock';
import mockShow from '../../mocks/show.mock';
import EpisodesList from './EpisodesList';

describe('<EpisodesList />', () => {
    it('should render EpisodesList', () => {
        const { container } = renderWithRouter(
            <EpisodesList episodes={mockShow.episodes} />,
            { route: `/shows/6771` }
        );

        expect(container).toMatchSnapshot();
    });
});
