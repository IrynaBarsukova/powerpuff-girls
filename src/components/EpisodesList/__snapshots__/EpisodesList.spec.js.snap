// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodesList /> should render EpisodesList 1`] = `
<div>
  <div
    class="episodes"
  >
    <h3
      class="episodes__title"
    >
      Episodes
    </h3>
    <div
      class="episodesList"
    >
      <div
        class="episodesList__episode"
      >
        <a
          href="/shows/6771/episodes/657308"
        >
          <img
            alt="Escape from Monster Island"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/53/132617.jpg"
          />
          <p>
            Escape from Monster Island
          </p>
          <p>
            1
            x
            1
          </p>
        </a>
      </div>
      <div
        class="episodesList__episode"
      >
        <a
          href="/shows/6771/episodes/691594"
        >
          <img
            alt="Bye Bye, Bellum"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/53/132631.jpg"
          />
          <p>
            Bye Bye, Bellum
          </p>
          <p>
            1
            x
            7
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
